crates/kernel_cmdline/src/lib.rs

```diff
@@ -138,9 +145,38 @@ impl<'a> std::ops::Deref for ParameterKey<'a> {
     }
 }
 
-impl<'a> From<&'a [u8]> for ParameterKey<'a> {
-    fn from(value: &'a [u8]) -> Self {
-        Self(value)
+impl<'a> TryFrom<&'a str> for ParameterKey<'a> {
+    type Error = <ParameterKey<'a> as TryFrom<&'a [u8]>>::Error;
+
+    /// Parses a parameter key from a UTF-8 string.
+    ///
+    /// Any leading or trailing ascii whitespace is trimmed.
+    ///
+    /// Returns an error if the key contains internal whitespace.
+    fn try_from(value: &'a str) -> Result<Self, Self::Error> {
+        Self::try_from(value.as_bytes())
+    }
+}
+
+impl<'a> TryFrom<&'a [u8]> for ParameterKey<'a> {
+    type Error = anyhow::Error;
+
+    /// Parses a parameter key from raw bytes.
+    ///
+    /// Any leading or trailing ascii whitespace is trimmed.
+    ///
+    /// Returns an error if the key contains internal whitespace.
+    fn try_from(value: &'a [u8]) -> Result<Self, Self::Error> {
```

Comment:
I think it might improve ergonomics to define this in terms of a function that returns `Option`, and have this one just use `.ok_or_else`.

The reason here is that the one that returns `Option` can be `const` and so since Rust 1.83 it should be possible to do e.g.:

```
let parameter = const { Parameter::from_bytes("foo").unwrap() }
```

So for constant strings like this if it's invalid we fail at compile time, not runtime.